brokers/mqtt/src/config.rs

```diff
@@ -1,45 +1,106 @@
-use std::{path::PathBuf, time::Duration};
-
+use anyhow::Result;
+use clap::Parser;
+use config;
 use humantime::parse_duration;
 use paho_mqtt::{
 	ConnectOptions, ConnectOptionsBuilder, CreateOptions, CreateOptionsBuilder, Error, SslOptions,
 	SslOptionsBuilder,
 };
-use structopt::StructOpt;
+use serde::{Deserialize, Serialize};
+use std::{path::PathBuf, time::Duration};
 
-#[derive(Debug, StructOpt)]
-#[structopt(name = "spectacles-mqtt", about = env!("CARGO_PKG_DESCRIPTION"))]
-pub struct Opt {
-	#[structopt(flatten)]
+#[derive(Debug, Serialize, Deserialize, Parser)]
+#[command(name = "spectacles-redis", about = env!("CARGO_PKG_DESCRIPTION"))]
+pub struct Config {
+	#[arg(long, short, env = "MQTT_CONFIG_FILE", exclusive = true)]
+	pub config_file: Option<String>,
+
+	#[command(flatten)]
+	#[serde(flatten)]
 	pub create: CreateOpt,
-	#[structopt(flatten)]
+
+	#[command(flatten)]
 	pub connect: ConnectOpt,
+
 	/// Events to subscribe to.
-	#[structopt(long)]
+	#[arg(short, long, env = "MQTT_EVENTS", value_delimiter = ',')]
+	#[serde(default = "Config::default_events")]
```

Comment:
I think I may have misled you a bit here; since `Vec` implements `Default`, we don't need to specify a custom default factory. This can be applied in a few other cases: `bool` defaults to `false`, `String` to `""`, etc. The associated default functions can be removed as well.

Sorry for the confusion!

```suggestion
	#[serde(default)]
```

Reply:
Ahh that makes more sense, ty for pointing that out, fixed it!